### PR TITLE
Moved OQS KEX definitions to the crypto layer

### PR DESCRIPTION
**Pull request by christianpaquin**
Moved OQS KEX definitions from the ssl layer to the crypto layer. OpenSSL's internal NIDs are now automatically generated through the build system (defined in `crypto/objects/objects.txt` and generate by `make generate_crypto_objects`) and are distinct from the TLS curve/group IDs (still defined by our own code in the ssl layer, because we define special code points for these "curves").

These changes should allow 3rd party applications (such as OpenVPN) to use the oqs algs when looking them up by NID.
`
This is a draft request:
- Some new code needs to be integrated with templating framework

### CODE REVIEW

**Comment by christianpaquin** [comment]
@xvzcf, could you check that my modifications to the templating files are ok?

**Comment by xvzcf** [comment]
The templating changes look fine to me.

File: ssl/t1_lib.c [hunk 1]
```diff
@@ -570,10 +571,10 @@ static int nid_cb(const char *elem, int len, void *arg)
...
     if (nid == NID_undef)
         nid = OBJ_ln2nid(etmp);
-    /* OQS note: parse oqs algs */
-    if (nid == NID_undef) {
-      nid = OQS_nid_from_string(etmp);
-    }
+    /* /\* OQS note: parse oqs algs *\/ */
+    /* if (nid == NID_undef) { */
+    /*   nid = OQS_nid_from_string(etmp); */
+    /* } */
```
Comment (dstebila) [inline ssl/t1_lib.c]: Can this code be deleted rather than just commented out?

**Review by dstebila — CHANGES_REQUESTED** [review]

File: ssl/t1_lib.c (hunk 1 above)
Comment (christianpaquin) [reply ssl/t1_lib.c]: Yes, I'll remove it.

**Review by dstebila — APPROVED** [review]

**Pull request merged**